prog/mutation.go

```diff
@@ -15,220 +15,228 @@ import (
 func (p *Prog) Mutate(rs rand.Source, ncalls int, ct *ChoiceTable, corpus []*Prog) {
 	r := newRand(rs)
 
-	if r.oneOf(100) && corpus != nil {
-		// Splice with another prog from corpus.
-		p0 := corpus[r.Intn(len(corpus))]
-		p0c := p0.Clone()
-		idx := r.Intn(len(p.Calls))
-		p.Calls = append(p.Calls[:idx], append(p0c.Calls, p.Calls[idx:]...)...)
-	} else {
-		// Mutate current prog without splicing.
-		retry := false
-		for stop := false; !stop || retry; stop = r.oneOf(3) {
-			retry = false
-			switch {
-			case r.nOutOf(20, 31):
-				// Insert a new call.
-				if len(p.Calls) >= ncalls {
-					retry = true
-					continue
-				}
-				idx := r.biasedRand(len(p.Calls)+1, 5)
-				var c *Call
-				if idx < len(p.Calls) {
-					c = p.Calls[idx]
-				}
-				s := analyze(ct, p, c)
-				calls := r.generateCall(s, p)
-				p.insertBefore(c, calls)
-			case r.nOutOf(10, 11):
-				// Change args of a call.
-				if len(p.Calls) == 0 {
-					retry = true
-					continue
-				}
-				c := p.Calls[r.Intn(len(p.Calls))]
-				if len(c.Args) == 0 {
+	retry := false
+	for stop := false; !stop || retry; stop = r.oneOf(3) {
+		retry = false
+		switch {
+		case r.nOutOf(1, 100):
+			// Splice with another prog from corpus.
+			if corpus == nil {
```

Comment:
s/corpus == nil/len(corpus) == 0/
A slice can be non-nil, but with len == 0:
https://play.golang.org/p/4hxVJV1OPb

Also, please merge the two conditions into a single if:

```
if len(corpus) == 0 || len(p.Calls) == 0 {
```